lib/ansible/modules/system/ufw.py

```diff
@@ -52,7 +52,34 @@
     choices: [ on, off, low, medium, high, full ]
   insert:
     description:
-      - Insert the corresponding rule as rule number NUM
+      - Insert the corresponding rule as rule number NUM.
+      - Note that ufw numbers rules starting with 1.
+    type: int
```

Comment:
No defaults? Why?

Reply:
Depending on what `insert_relative_to` is, the default should be 0 or 1. So it's better to not have a default, and in case this is not specified, select a sensible default depending on context in the code.

Reply:
I see you point from detailed code review. But what if we have two variables?
`insert`, type: `boolean` - defining do we need to insert the rule with default `false`
`insert_position`, type: `int` - insert as a specified rule number with default `0`

Reply:
And if you are using `insert` together with `insert_relative_to` then `insert_position` will be calculated automatically

Reply:
The problem is that `insert` has already been there, so we cannot change its type to `bool` without breaking backwards compatibility.

Reply:
But it was string type?

Reply:
OK, I buy it. No other easy way to keep compatibility.

Reply:
It was string type, but if you put in there anything else than an integer (in the correct range), `ufw` would die anyway. So forcing it to be of type `int` shouldn't break any valid use-case :)